Microsoft.Identity.Web/AzureADB2COpenIDConnectEventHandlers.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Web
+{
+    internal class AzureADB2COpenIDConnectEventHandlers
```

Comment:
B2C needs special OIDC handlers to cover the scenario where the user clicks on EditProfile. The issuer would have the "edit_profile" policy due to the `AuthenticationProperties` set on Microsoft.Identity.UI `EditProfile`, however it needs to use the default policy (sisu). 

It also handles the case when the user cancels an action on B2C UI.

These were all gotten from AzureADB2C